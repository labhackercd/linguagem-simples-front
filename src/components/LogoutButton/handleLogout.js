import axiosInstance from '../../auth/axiosApi'

export default function handleLogout() {
    try {
        localStorage.removeItem('access_token');
        localStorage.removeItem('refresh_token');
        axiosInstance.defaults.headers['Authorization'] = null;
    }
    catch (e) {
        //console.log(e);
    }
};