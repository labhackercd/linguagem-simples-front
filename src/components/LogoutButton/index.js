import axiosInstance from './../../auth/axiosApi.js'
import React from 'react';

import Button from '@material-ui/core/Button'
import { makeStyles } from '@material-ui/core/styles';
import handleLogout from './handleLogout.js'
import { ReactComponent as LogoutIcon } from './assets/logout_icon.svg';
import SvgIcon from '@material-ui/core/SvgIcon'
import {INITIAL_PAGE_URL} from '../../api_urls'

const useStyles = makeStyles({
    inputTextColor:{
        color:'white',
      },
    label: {
    textTransform: 'capitalize',
    }
  });



export default function LogoutButton() {
    const classes = useStyles();

    return (
        <Button
          onClick={() => { handleLogout() }}
          classes={{
              root: classes.inputTextColor, // class name, e.g. `classes-nesting-root-x`
              label: classes.label, // class name, e.g. `classes-nesting-label-x`
            }}
          href={INITIAL_PAGE_URL}
          endIcon={<LogoutIcon />}
          >
        sair
      </Button>
    );
  }
