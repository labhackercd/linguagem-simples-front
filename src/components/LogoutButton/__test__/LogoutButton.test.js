import React from 'react'
import {render} from '@testing-library/react';
import ReactDOM from 'react-dom'
import LogoutButton from './../index'
import renderer from 'react-test-renderer';



test('Test if LogoutButton renders without crash', () => {
    const div = document.createElement("div")
    ReactDOM.render(<LogoutButton></LogoutButton>, div)
    ReactDOM.unmountComponentAtNode(div)
});

test('Test if LogoutButton renders local text properly', () => {
    const { getByText } = render(<LogoutButton />);
    const linkElement = getByText(/sair/i);
    expect(linkElement).toBeInTheDocument();
});

test('Test if LogoutButton snapshot matches', () => {
    const component = renderer.create(
        <LogoutButton></LogoutButton>
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
