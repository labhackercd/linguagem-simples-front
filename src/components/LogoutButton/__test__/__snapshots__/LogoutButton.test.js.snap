// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if LogoutButton snapshot matches 1`] = `
<a
  aria-disabled={false}
  className="MuiButtonBase-root MuiButton-root makeStyles-inputTextColor-5 MuiButton-text"
  href="/"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
>
  <span
    className="MuiButton-label makeStyles-label-6"
  >
    sair
  </span>
</a>
`;
